search/scorer/scorer_knn.go

```diff
@@ -73,9 +71,36 @@ func (sqs *KNNQueryScorer) Score(ctx *search.SearchContext,
 			score = 1.0 / score
 		}
 
+		if sqs.options.Explain {
+			childrenExplanations := make([]*search.Explanation, 1)
+			childrenExplanations[0] = &search.Explanation{
+				Value: score,
+				Message: fmt.Sprintf("vector(field(%s:%s) with similarity_metric(%s)=%f",
+					sqs.queryField, knnMatch.ID, sqs.similarityMetric, score),
+			}
+			scoreExplanation = &search.Explanation{
+				Value: score,
+				Message: fmt.Sprintf("fieldWeight(%s in doc %s), score of:",
+					sqs.queryField, knnMatch.ID),
+				Children: childrenExplanations,
+			}
+		}
+
 		// if the query weight isn't 1, multiply
 		if sqs.queryWeight != 1.0 {
 			score = score * sqs.queryWeight
+			if sqs.options.Explain {
+				childExplanations := make([]*search.Explanation, 2)
+				childExplanations[0] = sqs.queryWeightExplanation
+				childExplanations[1] = scoreExplanation
+				scoreExplanation = &search.Explanation{
+					Value: score,
+					// Product of score * weight
+					Message: fmt.Sprintf("weight(%s:%f^%f in %s), product of:",
+						sqs.queryField, sqs.queryVector, sqs.queryBoost, knnMatch.ID),
```

Comment:
Just noticed this, let's not emit the actual query vector to the output - there'll be an overload of data.